dev-libs/boost/boost-1.88.0.ebuild

```diff
@@ -207,6 +205,108 @@ multilib_src_compile() {
+# TODO: multilib-minimal_abi_src_test is somehow not invoked
+src_test() {
+	# only use the best ABI
+	BUILD_DIR="${S}-$(echo $(multilib_get_enabled_abi_pairs) | awk '{print $NF}')"
+
+	# create custom options for tests based on the build settings
+	TEST_OPTIONS=("${OPTIONS[@]}")
+
+	# dial down log output - the full b2 command used to compile & run
+	# a test suite will be printed by ejam and can be used to build
+	# and run the tests in a test suite's directory.
+	TEST_OPTIONS=("${TEST_OPTIONS[@]/-d+2/-d0}")
+
+	# The following libraries do not compile or fail their tests:
+	# algorithm: fails to use std::reverse_copy (std=c++20?)
+	# config: test output comparison failure
+	# contract: "C++03 support was deprecated in Boost.Chrono 1.82" ??
+	# geometry: passes but takes >1 hour to build
+	# graph_parallel: tries to use mpi even with use=-mpi
+	# math: undefined reference to `boost::math::concepts::real_concept boost::math::bernoulli_b2n<boost::math::concepts::real_concept>(int)
+	# multiprecision: error: assignment of read-only member 'gauss::laguerre::detail::laguerre_l_object<T>::order'
+	# mysql:  error: 'uint8_t' is not a member of 'std'
+	# parameter_python: PyObject* boost::parameter::python::aux::unspecified_type():
+	#   /usr/include/python3.13/object.h:339:30: error: lvalue required as left operand of assignment
+	# phoenix: scope/lambda_tests22.cpp(27): test 'x == 1' failed in function 'int main()'
+	# predef: error: Unable to find file or target named (yes, really)
+	# python: AttributeError: property '<unnamed Boost.Python function>' of 'X' object has no setter
+	# qvm: vec_access.hpp:95:223: error: static assertion failed: Boost QVM static assertion failure
+	# regex: regex_timer.cpp:19: ../../../boost/timer.hpp:21:3: error: #error This header is
+	#   deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress
+	#   this error.)
+	# serialization: in function `boost::archive::tmpnam(char*)': test_array.cpp:(.text+0x108):
+	#   undefined reference to `boost::filesystem::detail::unique_path(...)'
+	# statechart: TuTestMain.cpp(22) fatal error: in "test_main_caller( argc_ argv )":
+	#   std::runtime_error: Event was not consumed!
+	# unordered: erase_tests.cpp:(.text+0x44cce): undefined reference to
+	#   tbb::detail::r1::execution_slot(tbb::detail::d1::execution_data const*)
+	# wave: t_5_007.cpp(22): error: could not find include file: boost/version.hpp
+	local libs_excluded=(
+		"algorithm" "config" "contract" "geometry" "graph_parallel" "math"
+		"multiprecision" "mysql" "parameter_python" "phoenix" "predef"
+		"python" "qvm" "regex" "serialization" "statechart" "unordered"
+		"wave"
+	)
+
+	if use test-full; then
+		ewarn "Enabling expensive tests - this will take a long time!"
+		libs_excluded=("${libs_excluded[@]/geometry}")
+	fi
+
+	# libraries which must be patched with an additonal main()
+	local libs_needpatch=(
+		"accumulators"
+	)
+
+	# prepare to find libraries without failing candidates
```

Comment:
This whole way of finding the directory names by creating a "find" command with exclusions was the best I could think of. If there is a better way I'd love to hear it.